### DISCUSSION TITLE
v0.3.6 - nim 2.0 compatibility!

### DISCUSSION THREAD

**Discussion by pietroppeter** [pietroppeter/nimib#175] [Announcements]
## What's Changed

* Change toml lib + fix macro bug on devel by @HugoGranstrom in https://github.com/pietroppeter/nimib/pull/173
  with this PR nimib is finally compatible with the upcoming 2.0. Two things needed fixing:
  - a nim compiler orc macro bug (reported here: https://github.com/nim-lang/Nim/issues/21326), fixed with a [workaround]
  - there was an issue with toml serialization on orc (reported here: https://github.com/status-im/nim-toml-serialization/issues/62) fixed by removing the dependency from [toml_serialization] and using [parsetoml] instead
    - some custom code in order to support direct to object conversion was needed, see https://github.com/NimParsers/parsetoml/issues/59
    - we also added a dependency to [jsony] to support a loose direct to object conversion from toml
* Adding devel to CI, fix #161, fix #149 by @pietroppeter in https://github.com/pietroppeter/nimib/pull/169
  - added both stable (which will become 2.0) and devel (which is currently the 2.0 rc) to CI, also update the versions of actions we depend on

**Full Changelog**: https://github.com/pietroppeter/nimib/compare/v0.3.5...v0.3.6

[workaround]: https://github.com/pietroppeter/nimib/pull/173/commits/f1966097da91d4ba7b2711dbe44223e871927b42
[toml_serialization]: https://github.com/status-im/nim-toml-serialization
[parsetoml]: https://github.com/NimParsers/parsetoml
[jsony]: https://github.com/treeform/jsony

<hr /><em>This discussion was created from the release <a href='https://github.com/pietroppeter/nimib/releases/tag/v0.3.6'>v0.3.6 - nim 2.0 compatibility!</a>.</em>

**Comment by pietroppeter** [comment]
great job @HugoGranstrom on this one, found a few nasty bugs along the way! 👏

**Reply by pietroppeter** [reply]
this deserved a discussion also because it is progress on our 2023H1 goals set last month! https://github.com/pietroppeter/nimib/milestone/1

**Reply by HugoGranstrom** [reply]
I must say that the milestones are very nice as you can see the progress and a nice list of issues :D
